<launch>
  <arg name="ws" default="$(find rsim_kmt)" />
  <rosparam command="load" file="$(arg ws)/recipe/param.yaml" />
<!-- <rosparam command="load" file="$(arg ws)/rcalib.yaml" /> -->
  <node pkg="rsim_kmt" type="searcher2d.py" name="" output="screen">
    <remap from="~config" to="/config/sercher2d" />
    <remap from="~param" to="/searcher2d" />
    <remap from="~image_in" to="/rovi/left/image_rect" />
    <remap from="~do1" to="/request/capture" />
    <remap from="~do2" to="/request/solve" />
    <remap from="~done1" to="/response/capture" />
    <remap from="~done2" to="/response/solve" />
    <remap from="~thru1" to="/searcher2d/thru1" />
    <remap from="~thru2" to="/searcher2d/thru2" />
  </node>
  <node pkg="rovi_utils" type="cropper.py" name="surface_cropper" output="screen">
    <remap from="~config" to="/config/cropper" />
    <remap from="~param" to="/cropper" />
    <remap from="~update" to="/update/cropper" />
    <remap from="~capture" to="/searcher2d/thru1" />
    <remap from="~captured" to="/response/capture" />
    <remap from="~clear" to="/request/clear" />
    <remap from="~redraw" to="/request/redraw" />
    <remap from="~in/floats" to="/rovi/ps_floats" />
    <remap from="~out/floats" to="/surface/cropped/floats" />
    <remap from="~raw/floats" to="/scene/surface/raw/floats" />
  </node>
 <node pkg="rovi_utils" type="searcher.py" name="searcher0" args="path:=$(find rovi_visual_teach)/recipe" output="screen">
    <remap from="~config" to="/config/searcher0" />
    <remap from="~param" to="/searcher" />
    <remap from="~in/surface/floats" to="/scene/surface/floats" />
    <remap from="~master/surface/floats" to="/master/surface/floats" />
    <remap from="~clear" to="/request/clear" />
    <remap from="~load" to="/request/model_load" />
    <remap from="~loaded" to="/response/recipe_load" />
    <remap from="~save" to="/request/model_save" />
    <remap from="~saved" to="/response/model_save" />
    <remap from="~solve" to="/searcher2d/thru2" />
    <remap from="~solved" to="/searcher/solved" />
    <remap from="~redraw" to="/request/redraw" />
    <remap from="~score" to="/searcher/score" />
    <remap from="~stat" to="/searcher/stat" />
  </node>
  <node pkg="rovi_utils" type="picker.py" name="" output="screen">
    <remap from="~config" to="/config/picker" />
    <remap from="~param" to="/picker" />
    <remap from="~clear" to="/request/clear" />
    <remap from="~solve" to="/searcher/solved" />
    <remap from="~score" to="/searcher/score" />
    <remap from="~solved" to="/response/solve" />
    <remap from="~redraw" to="/request/redraw" />
  </node>
  <node pkg="rovi_utils" type="mesh_aid.py" name="mesh_aid" output="screen">
    <remap from="~config" to="/config/mesh_aid" />
    <remap from="~preset" to="/mesh_aid/preset" />
    <remap from="~solver_param" to="/searcher" />
  </node>
</launch>
